src/Factories/RoundFactory.php

```diff
@@ -21,17 +21,27 @@ class RoundFactory
      * @param int $playersPerRace
      * @param bool $mixedAlignment
      * @return Round
+     * @throws \Exception
      */
     public function create(RoundLeague $league, Carbon $startDate, int $realmSize, int $packSize, int $playersPerRace, bool $mixedAlignment): Round
     {
         $number = ($this->getLastRoundNumber($league) + 1);
+        $endDate = (clone $startDate)->addDays(static::ROUND_DURATION_IN_DAYS);
+
+        $invasionEndHours = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 12, 13, 14, 15, 16, 12, 13, 14, 15, 16];
+        shuffle($invasionEndHours);
+
+        $hoursBeforeRoundEnd = $invasionEndHours[random_int(0, count($invasionEndHours) - 1)];
+
+        $offensiveActionsEndDate = (clone $endDate)->addHours(-$hoursBeforeRoundEnd);
 
         return Round::create([
             'round_league_id' => $league->id,
             'number' => $number,
             'name' => "Beta Round {$number}", // todo
             'start_date' => $startDate,
-            'end_date' => (clone $startDate)->addDays(static::ROUND_DURATION_IN_DAYS),
+            'end_date' => $endDate,
+            'offensive_actions_end_date' => $offensiveActionsEndDate,
```

Comment:
Don't forget to change column name (and variable name) from feedback in migration up above.